### PR TITLE
Don't check XR compat flag for inline sessions

### PR DESCRIPTION
**Pull request by Manishearth**
fixes https://github.com/immersive-web/webxr/issues/607

r? @NellWaliczek

### CODE REVIEW

**Comment by Manishearth** [comment]
I don't think this is the complete solution. This doesn't address the case where there are multiple inline sessions present, since it invalidates the other inline sessions. A fix for that is to _only_ unset the flag for `baseLayer`, but that doesn't solve the issue of creating a compatible context and using it for a different device later.

It feels like `makeXRCompatible()` should take a session parameter and mark the context as compatible for that session only.

Alternatively, we can make inline sessions skip this flag completely. I might just do that.

**Comment by Manishearth** [comment]
> Alternatively, we can make inline sessions skip this flag completely. I might just do that.

This was cleaner, ended up doing it.

cc @asajeffrey

**Comment by toji** [comment]
I don't think this is the right approach. As long as the underlying XR device hasn't changed there's no reason that the compatibility bit would become invalid between one XR session and the next. The only time that this would be problematic is if the XR device is changed, and that should already be covered by the algorithm here: http://localhost:8080/webxr/#select-an-xr-device

In retrospect I'm not sure why we left #607 open, as it likely could have been easily explained and closed.

**Comment by Manishearth** [comment]
In that case should I keep the portion of this PR that ignores the flag for inline sessions?

**Comment by Manishearth** [comment]
Since that's a problem regardless of this.

**Comment by toji** [comment]
Editors talked this over and we do want to keep the text about ignoring the flag for inline sessions, but we feel the current context compatibility set/unset logic is correct and should not be changed. If you can refactor the PR to only include the change to the inline check, we'll review it again. Thanks!

**Comment by Manishearth** [comment]
Done!

File: index.bs [hunk 1]
```diff
@@ -1605,7 +1605,7 @@ The <dfn constructor for="XRWebGLLayer">XRWebGLLayer(|session|, |context|, |laye
   1. Let |layer| be a new {{XRWebGLLayer}}
   1. If |session|'s [=ended=] value is <code>true</code>, throw an {{InvalidStateError}} and abort these steps.
   1. If |context| is lost, throw an {{InvalidStateError}} and abort these steps.
-  1. If |context|'s [=XR compatible=] boolean is false, throw an {{InvalidStateError}} and abort these steps.
+  1. If |context|'s [=XR compatible=] boolean is false and |session|'s [=XRSession/mode=] is not {{XRSessionMode/"inline"}}, throw an {{InvalidStateError}} and abort these steps.
```
Comment (toji) [inline index.bs]: Let's change this to use the [=immersive session=] definition instead, which should be a bit more future-proof.

> If |session| is an [=immersive session=] and |context|'s [=XR compatible=] boolean is &lt;code&gt;false&lt;/code&gt;, throw an {{InvalidStateError}} and abort these steps.

**Comment by Manishearth** [comment]
:rofl: 

done!

**Comment by toji** [comment]
👍

**Pull request merged**